lua/entities/gmod_wire_characterlcd/cl_init.lua

```diff
@@ -222,20 +225,249 @@ local specialCharacters = {
+  },
+  [148] = {
+    { x = 0.33, y = 0},
+    { x = 0.66, y = 0},
+    { x = 0.66, y = 1},
+    { x = 0.33, y = 1},
+  },
+  [149] = {
+    { x = 0.66, y = 0.66 },
+    { x = 0, y = 0.66 },
+    { x = 0, y = 0.33 },
+    { x = 0.33, y = 0.33 },
+    { x = 0.33, y = 0 },
+    { x = 0.66, y = 0 },
+  },
+  [150] = {
+    { x = 0, y = 0.33},
+    { x = 1, y = 0.33},
+    { x = 1, y = 0.66},
+    { x = 0, y = 0.66},
+  },
+  [151] = {
+    { x = 0.66, y = 0.33 },
+    { x = 1, y = 0.33 },
+    { x = 1, y = 0.66 },
+    { x = 0, y = 0.66 },
+    { x = 0, y = 0.33 },
+    { x = 0.33, y = 0.33 },
+    { x = 0.33, y = 0 },
+    { x = 0.66, y = 0 },
+  },
+  [152] = {
+    { x = 0.66, y = 0.33 },
+    { x = 1, y = 0.33 },
+    { x = 1, y = 0.66 },
+    { x = 0.66, y = 0.66 },
+    { x = 0.66, y = 1 },
+    { x = 0.33, y = 1 },
+    { x = 0.33, y = 0 },
+    { x = 0.66, y = 0 },
+  },
+  [153] = {
+    { x = 0.66, y = 0.66 },
+    { x = 1, y = 0.66 },
+    { x = 1, y = 0.33 },
+    { x = 0, y = 0.33 },
+    { x = 0, y = 0.66 },
+    { x = 0.33, y = 0.66 },
+    { x = 0.33, y = 1 },
+    { x = 0.66, y = 1 },
+  },
+  [154] = {
+    { x = 0.33, y = 0.33 },
+    { x = 0, y = 0.33 },
+    { x = 0, y = 0.66 },
+    { x = 0.33, y = 0.66 },
+    { x = 0.33, y = 1 },
+    { x = 0.66, y = 1 },
+    { x = 0.66, y = 0 },
+    { x = 0.33, y = 0 },
+  },
+  [155] = {
+    { x = 0.66, y = 0.33 },
+    { x = 1, y = 0.33 },
+    { x = 1, y = 0.66 },
+    { x = 0.66, y = 0.66 },
+    { x = 0.66, y = 1 },
+    { x = 0.33, y = 1 },
+    { x = 0.33, y = 0.66 },
+    { x = 0, y = 0.66 },
+    { x = 0, y = 0.33 },
+    { x = 0.33, y = 0.33 },
+    { x = 0.33, y = 0 },
+    { x = 0.66, y = 0 },
+  },
+  [156] = {
+    { x = 0.33, y = 0 },
+    { x = 0.66, y = 0 },
+    { x = 0.66, y = 0.33 },
+    { x = 0.33, y = 0.33 },
+  },
+  [157] = {
+    { x = 0.66, y = 0.33 },
+    { x = 1, y = 0.33 },
+    { x = 1, y = 0.66 },
+    { x = 0.66, y = 0.66 },
+  },
+  [158] = {
+    { x = 0.33, y = 0.66 },
+    { x = 0.66, y = 0.66 },
+    { x = 0.66, y = 1 },
+    { x = 0.33, y = 1 },
+  },
+  [159] = {
+    { x = 0, y = 0.33 },
+    { x = 0.33, y = 0.33 },
+    { x = 0.33, y = 0.66 },
+    { x = 0, y = 0.66 },
+  },
+  [160] = {
+    { x = 0.33, y = 0.33 },
+    { x = 0.66, y = 0.33 },
+    { x = 0.66, y = 0.66 },
+    { x = 0.33, y = 0.66 },
+  }
 }
 
 function ENT:DrawSpecialCharacter(c,x,y,w,h,r,g,b)
   surface.SetDrawColor(r,g,b,255)
   surface.SetTexture(0)
 
-  local vertices = specialCharacters[c]
+  local vertices = table.Copy(specialCharacters[c])
   if vertices then
-    local vertexData = {
+    --[[local vertexData = {
       { x = vertices[1].x*w+x, y = vertices[1].y*h+y },
       { x = vertices[2].x*w+x, y = vertices[2].y*h+y },
       { x = vertices[3].x*w+x, y = vertices[3].y*h+y },
-    }
-    surface.DrawPoly(vertexData)
+    }]]
+    for i=1,#vertices do
+      vertices[i].x = vertices[i].x*w+x
+      vertices[i].y = vertices[i].y*h+y
+    end
```

Comment:
Could you try replacing the table.Copy and math stuff here with 
```lua
local tf = Matrix() tf:SetScale(Vector(w, h, 1)) tf:SetTranslation(x,y)
cam.PushModelMatrix(tf, true)
surface.DrawPoly(vertices)
cam.PopModelMatrix()
```

Reply:
Done, i think it works. Thanks!